token/services/interop/htlc/wallet_filter.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package htlc
+
+import (
+	"bytes"
+	"encoding/base64"
+	"time"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/view"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/token"
+	"go.uber.org/zap/zapcore"
+)
+
+// PickFunction is a prototype for (token,script) pair selection
+type PickFunction = func(*token.UnspentToken, *Script) (bool, error)
+
+type PreImageFilter struct {
+	preImage []byte
+}
+
+func (f *PreImageFilter) Filter(tok *token.UnspentToken, script *Script) (bool, error) {
```

Comment:
Why does the function takes tok. It seems that tok is only used for logging purposes.

Reply:
You are right, I left it just for the future :)